maas/resource_maas_boot_source_selection.go

```diff
@@ -143,12 +171,21 @@ func resourceBootSourceSelectionUpdate(ctx context.Context, d *schema.ResourceDa
 		return diag.FromErr(err)
 	}
 
+	if err := client.BootResources.Import(); err != nil {
+		return diag.FromErr(err)
+	}
+
 	return resourceBootSourceSelectionRead(ctx, d, meta)
 }
 
 func resourceBootSourceSelectionDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*ClientConfig).Client
 
+	err := awaitImportComplete(client)
```

Comment:
I think that this import is not needed here. Instead, we may have to run it after deleting the selection so that MAAS can clear the boot resources of this selection.